cmd/soroban-rpc/internal/jsonrpc.go

```diff
@@ -128,89 +128,117 @@ func NewJSONRPCHandler(cfg *config.Config, params HandlerParams) Handler {
+			underlyingHandler:    methods.NewSendTransactionHandler(params.Daemon, params.Logger, params.TransactionStore, cfg.NetworkPassphrase),
+			longName:             "send_transaction",
+			queueLimit:           cfg.RequestBacklogSendTransactionQueueLimit,
+			requestDurationLimit: cfg.MaxSendTransactionExecutionDuration,
 		},
 		{
-			methodName:        "simulateTransaction",
-			underlyingHandler: methods.NewSimulateTransactionHandler(params.Logger, params.LedgerEntryReader, params.PreflightGetter),
-			gaugeName:         "simulate_transaction_inflight_requests",
-			gaugeHelp:         "Number of concurrenty in-flight simulateTransactions requests",
-			queueLimit:        cfg.RequestBacklogSimulateTransactionQueueLimit,
+			methodName:           "simulateTransaction",
+			underlyingHandler:    methods.NewSimulateTransactionHandler(params.Logger, params.LedgerEntryReader, params.PreflightGetter),
+			longName:             "simulate_transaction",
+			queueLimit:           cfg.RequestBacklogSimulateTransactionQueueLimit,
+			requestDurationLimit: cfg.MaxSimulateTransactionExecutionDuration,
 		},
 	}
 	handlersMap := handler.Map{}
 	for _, handler := range handlers {
-		gauge := prometheus.NewGauge(prometheus.GaugeOpts{
+		queueLimiterGaugeName := handler.longName + "_inflight_requests"
+		queueLimiterGaugeHelp := "Number of concurrenty in-flight " + handler.methodName + " requests"
+
+		queueLimiterGauge := prometheus.NewGauge(prometheus.GaugeOpts{
 			Namespace: params.Daemon.MetricsNamespace(), Subsystem: "network",
-			Name: handler.gaugeName,
-			Help: handler.gaugeHelp,
+			Name: queueLimiterGaugeName,
+			Help: queueLimiterGaugeHelp,
 		})
-		limiter := network.MakeJrpcBacklogQueueLimiter(
+		queueLimiter := network.MakeJrpcBacklogQueueLimiter(
 			handler.underlyingHandler,
-			gauge,
+			queueLimiterGauge,
 			uint64(handler.queueLimit),
 			params.Logger)
-		handlersMap[handler.methodName] = limiter
+
+		durationWarnCounterName := handler.longName + "_execution_threshold_warning"
+		durationLimitCounterName := handler.longName + "_execution_threshold_limit"
+		durationWarnCounterHelp := "The metric measures the count of " + handler.methodName + " requests that surpassed the warning threshold for execution time"
+		durationLimitCounterHelp := "The metric measures the count of " + handler.methodName + " requests that surpassed the limit threshold for execution time"
+
+		requestDurationWarnCounter := prometheus.NewCounter(prometheus.CounterOpts{
+			Namespace: params.Daemon.MetricsNamespace(), Subsystem: "network",
+			Name: durationWarnCounterName,
+			Help: durationWarnCounterHelp,
+		})
+		requestDurationLimitCounter := prometheus.NewCounter(prometheus.CounterOpts{
+			Namespace: params.Daemon.MetricsNamespace(), Subsystem: "network",
+			Name: durationLimitCounterName,
+			Help: durationLimitCounterHelp,
+		})
+		// set the warning threshold to be one third of the limit.
+		requestDurationWarn := handler.requestDurationLimit / 3
```

Comment:
I was kind of expecting `requestDurationWarn = request-execution-warning-threshold` to be used as that config setting seemed more aligned for warning?

Reply:
in order to reduce the total number of configuration options, I have included only a configuration for the *duration limit* of all the jrpc handlers. Then, I set a third of that to be the warning time. ( i.e. if it takes more then 30% of the max time, it would considered to be a warning ).

Reply:
oh, further reading below, `request-execution-warning-threshold` is meant for only the global rpc duration limit and then each endpoint dynamically creates it's specific warning by taking a 33% of it's duration max limit? Warnings could be factored into the [config types suggestion](https://github.com/stellar/soroban-tools/pull/804/files#r1278024068) also with a new suffix on the type of:

`-execution-duration-warning`

Reply:
The warning and limit are kind-of related. If we see an uptick in warning, it means that we need to do something before the limit would get hit. ( keep in mind that the two are exclusive : a handler that was `limit`-ed would not be considered `warn`-ed ).
The goal is that in case the server is gradually getting higher load, we would start getting warnings about that before the user would see any service disruption.
I don't mind removing the `RequestExecutionWarningThreshold` and making it a percentage of the `MaxRequestExecutionDuration` ( let's say 33% as well ).

Reply:
makes sense, I think we may need to consider adding a MONITORING.md to assist with expectations, the insights you're mentioning are valuable and sound like best practices/recommendations. i.e. use the defaults provided to start with and then the strategy for changing over time?

Reply:
makes sense, I think we may need to consider adding a MONITORING.md to assist with expectations, the insights you're mentioning are valuable and sound like best practices/recommendations. i.e. use the defaults provided to start with and then the strategy for changing over time?

Reply:
Yes, I think that adding that would be a great idea. I'll try to come up with one.

Reply:
@sreuland I added a MONITORING.md file. I'm sure that you'll have comments ( which are welcome! ) - could we defer the comments and proposed changes to a separate PR ?